### ISSUE TITLE
Can pdfrw replace image in pdf?

### ISSUE THREAD

**Issue by github163sl** [pmaupin/pdfrw#166]
Thank you！

**Comment by towr** [comment]
Yes, it can. It can be a bit tricky because there are a lot of different image types that can be embedded in a PDF in various ways. A JPEG (an image object with /Filter = /DCTDecode) can be very easily replaced by replacing the stream of the image object with the binary data of the new JPEG.

**Comment by abogaard** [comment]
Hi @towr would you mind please pointing us in the right direction of the image object stream? Within the pdfreader object, where does it reside?

**Comment by abogaard** [comment]
@towr would you please point us in the right direction with regards to replacing the string? I've got a DCTDecode object in a PDF and I am trying to replace the binary stream like: 

`template_pdf.pages[1]['/Resources']['/XObject']['/Im1'].stream = data`

But am getting an error (Error reading a stream) when I load the pdf. 'data' above is simply binary read from another jpg.

**Comment by towr** [comment]
It's been a while since I worked on this. But I think the two important things are to set the `...['/Im1'].Length` to the new length and to decode the data `data = data.decode('latin1')` (because assigning it encodes it, iirc)

Also make sure the colorspace of the JPEG is correct. (In particular PDFs may often have CMYK colorspace, whereas JPEGs you usually encounter are often RGB ;  I can't remember if I had trouble changing the `/Colorspace` value in the PDF, or just decided to keep things the same as much as possible.)

**Comment by abogaard** [comment]
Thanks for the thoughtful reply. I did play around with the meta tags a bit, but I'm not sure I hit the right combo. I'll have some time to work on this later today and I'll post back for anyone else who may be interested!
